streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraphBuilder.java

```diff
@@ -41,12 +45,16 @@ public JobGraph build() {
   }
 
   private void processStream(Stream stream) {
+    while (stream.isReferenceStream()) {
+      LOG.info("Skip reference stream {} of id {}", stream, stream.getId());
```

Comment:
What does "skip" means, still the control follow go through with `stream = stream. getReferencedStream()`

Reply:
`ReferenceStream` and `referenced stream` are the same logical stream, both refers to the same dataflow transformation. We skip `ReferenceStream` to avoid applying same transformation multiple times.

Reply:
I think this line log can be `debug`, user shouldn't care about this

Reply:
Added comments here